shelley/chain-and-ledger/executable-spec/src/Keys.hs

```diff
@@ -66,3 +86,27 @@ verifyKES (VKey vk) vd (KESig (Sig sd sk) m) n = vk == sk && vd == sd && m == n
 instance BA.ByteArrayAccess VKey where
   length        = BA.length . BS.pack . show
   withByteArray = BA.withByteArray . BS.pack . show
+
+instance BA.ByteArrayAccess VKeyGenesis where
+  length (VKeyGenesis vk)        = (BA.length . BS.pack . show) vk
+  withByteArray (VKeyGenesis vk) = (BA.withByteArray . BS.pack . show) vk
+
+newtype Dms = Dms (Map.Map VKeyGenesis VKey)
+  deriving (Show, Eq)
+
+newtype GKeys = GKeys (Set.Set VKeyGenesis)
+  deriving (Show, Eq)
+
+keyByronToShelley :: Byron.VKey -> VKey
```

Comment:
Maybe silly question, but why do we need these conversion functions? Can't we just use the same `VKey`/`VKeyGenesis` keys in both executable specs?

Reply:
yes, we could (the definitions are the same anyways)
Would you prefer that ?

Reply:
Ohhhh, I see, it has to do with the fact that we use `Word64` in Byron. 

Would it make sense to make an issue to unify these keys into a single type?

Reply:
slot is slightly different, the Byron spec uses `Word64`, Shelley `Natural`

Reply:
the idea was mainly to future-proof this for the case that something might differ for some reason, then only the conversion function would have to be adapted. In the best case it could just become `id`

Reply:
Hmm, normally I'd prefer not to introduce abstractions too early, even if it is a conversion function, since probably YAGNI... but I also understand this, so I trust your judgement :)

Reply:
> Would you prefer that ?

 I'm ok with both approaches. I'd prefer to write as little code as possible, but I wonder also what do we loose when concretizing the type to a `Word64`. So again, whatever you prefer....